awx/api/views/mixin.py

```diff
@@ -255,6 +257,8 @@ def get_serializer_context(self, *args, **kwargs):
                 org_reference = JT_inventory_reference
             elif res == 'users':
                 org_reference = 'id'
+            elif res == 'hosts':
+                org_reference = 'inventory__organization'
```

Comment:
oof, I know this logic is hard to understand, but looks like you extended it correctly. All of this count logic would be better if moved to the managers, but that's one of those things I thought of and never returned to. Implementing the new field via a manager method is great.